pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -264,6 +265,25 @@ func (c *Cluster) makeDeployment(osdProps osdProperties, osd OSDInfo) (*apps.Dep
 			{Name: "CEPH_VOLUME_DEBUG", Value: "1"},
 		}...)
 
+	} else if osd.IsDirectory {
+		command = []string{"ceph-osd"}
+		//args = opspec.DaemonFlags(c.ClusterInfo, osdID),
+		args = opconfig.DefaultFlags(c.clusterInfo.FSID, path.Join(osd.DataPath, "keyring"), c.clusterInfo.CephVersion)
+		args = append(args,
+			"--foreground",
+			"--id", osdID,
+			"--setuser", "ceph",
+			"--setgroup", "ceph",
+			"--osd-objectstore", "filestore", /* always? */
+			"--osd-max-object-name-len", "256",
+			"--osd-max-object-namespace-len", "64",
+			"--osd-journal-size", fmt.Sprintf("%d", osdProps.storeConfig.JournalSizeMB), /* will this get default value if user doesn't specify */
+			// public-addr ???
+			"--mon-initial-members", // TODO: get from StoredMonHostEnvVarFlags
+		)
+
+		// Set config in mon cfg database
+		// crush_location
```

Comment:
This PR is not done, and probably doesn't work at all at this point, but I have some questions for @leseb while I am working.
- `crush_location` - my thought is to set this in the mon config DB for each OSD since the user might want to override it at runtime. Or do you think it would be better to set this with a CLI flag?
- `osd-max-object-{name,namespace}-len` - my thought is that these probably shouldn't be overridden at runtime and therefore should be set by CLI flag. Do you agree?
- `osd-journal-size` - I am not sure if this should be constant (set via CLI) or whether this should be able to be overridden at runtime via the mon config DB. What is your thought?
- `public-addr` - does the OSD need this param provided to it? I'm really not sure.

Reply:
> This PR is not done, and probably doesn't work at all at this point, but I have some questions for @leseb while I am working.
> 
> * `crush_location` - my thought is to set this in the mon config DB for each OSD since the user might want to override it at runtime. Or do you think it would be better to set this with a CLI flag?

It makes sense because currently this can not be overridden by users, the CLI flag will always take precedence. There is just one thing I haven't tested and I'm doubting. Typically, to move an OSD to a different hierarchy, the admin would run `ceph osd crush move ... ...`. If we set something in the mon config store, I'm afraid the OSD will move back to that location. So TBT.

> * `osd-max-object-{name,namespace}-len` - my thought is that these probably shouldn't be overridden at runtime and therefore should be set by CLI flag. Do you agree?

Agreed, let's remove this.

> * `osd-journal-size` - I am not sure if this should be constant (set via CLI) or whether this should be able to be overridden at runtime via the mon config DB. What is your thought?

Not needed when the OSD starts, only useful for `prepare`.

> * `public-addr` - does the OSD need this param provided to it? I'm really not sure.
Currently, OSDs on c-v runs well without this so we can probably remove it.